### PR TITLE
feat: Causal Graph Matching

### PR DESCRIPTION
**Pull request by lmondada**

### CODE REVIEW

File: quizx/src/flow/causal.rs [hunk 1]
```diff
@@ -132,10 +133,41 @@ impl CausalFlow {
...
+
+    /// Whether u -> v is a causal edge.
+    pub fn is_causal_edge(&self, u: V, v: V) -> bool {
+        self.next(u) == Some(v)
+    }
+
+    /// Whether (u, v) is a non-causal edge.
+    ///
+    /// This is true when both u -> v and v -> u are not causal.
+    pub fn is_non_causal_edge(&self, u: V, v: V) -> bool {
```
Comment (aborgna-q) [inline quizx/src/flow/causal.rs]: A bit weird that `¬ (P ∧ ¬P ⊸ ⊤)` :P

Comment (lmondada) [reply quizx/src/flow/causal.rs]: true, feel free to suggest another name for "non-causal" edges :)

**Comment by lmondada** [comment]
This code should now work btw

File: quizx/src/flow/causal.rs (hunk 1 above)
Comment (lmondada) [reply quizx/src/flow/causal.rs]: how about I rename `is_causal` to `is_causal_dir` and then `is_causal` is true if the edge is causal in either direction

**Pull request merged**